File: SIMAR/Python/GNodo.py
# GNodo.py

import tkinter as tk

class GNodo:
    def __init__(self, x1, y1, id_nodo, seleccionado=False):
        self.x = x1
        self.y = y1
        self.ancho = 35
        self.id_nodo = id_nodo
        self.seleccionado = seleccionado
        self.circulo = None
        self.texto = None

    def getPoint(self):
        return (self.x, self.y)

    def setSelected(self, b):
        self.seleccionado = b

    def dibujar(self, canvas):
        if self.circulo:
            canvas.delete(self.circulo)
        if self.texto:
            canvas.delete(self.texto)

        color_borde = "red" if self.seleccionado else "blue"
        self.circulo = canvas.create_oval(self.x, self.y, self.x + self.ancho, self.y + self.ancho,
                                           outline=color_borde, width=3, fill="yellow")
        self.texto = canvas.create_text(self.x + self.ancho//2, self.y + self.ancho//2,
                                         text=self.id_nodo, font=("Arial", 12, "bold"))

    def dibujarUnselectedNodo(self, canvas):
        self.seleccionado = False
        self.dibujar(canvas)

    def dibujarSelectedNodo(self, canvas):
        self.seleccionado = True
        self.dibujar(canvas)

    def getId(self):
        return self.id_nodo

    def getCenter(self):
        return (self.x + self.ancho//2, self.y + self.ancho//2)

    def getMinX(self):
        return self.x

    def getMinY(self):
        return self.y

    def getMaxX(self):
        return self.x + self.ancho

    def getMaxY(self):
        return self.y + self.ancho

    def set(self, x, y):
        self.x = x
        self.y = y

    def isSelected(self):
        return self.seleccionado

    def contains(self, xi, yi):
        cx = self.x + self.ancho//2
        cy = self.y + self.ancho//2
        radio = self.ancho//2
        return ((xi - cx)**2 + (yi - cy)**2) <= radio**2

    def setId(self, c):
        self.id_nodo = c

File: SIMAR/GNodo.java
package SIMAR;

import java.awt.BasicStroke;
import java.awt.Color;
import java.awt.Graphics2D;
import java.awt.Point;
import java.awt.Stroke;
import java.awt.geom.Ellipse2D;

public class GNodo {
    private Stroke pincel;
    private Ellipse2D.Double path;
    private boolean selected;
    private char id;
    
    public GNodo(int x1, int y1, char id, boolean s) {
        this.path = new Ellipse2D.Double(x1, y1, 35.0D, 35.0D);
        this.id = id;
        this.selected = true;
        this.pincel = new BasicStroke(6.0F, 1, 0);
    }
    
    public Point getPoint() {
        int x = (int)this.path.getMinX();
        int y = (int)this.path.getMinY();
        return new Point(x, y);
    }
    
    public void setSelected(boolean b) {
        this.selected = b;
    }
    
    public void dibujarUnselectedNodo(Graphics2D gr2d) {
        gr2d.setStroke(this.pincel);
        gr2d.setColor(Color.BLUE);
        gr2d.draw(this.path);
        gr2d.setColor(Color.YELLOW);
        gr2d.fillOval((int)this.path.getMinX() + 4, (int)this.path.getMinY() + 4, 28, 28);
        gr2d.setColor(Color.BLACK);
        gr2d.drawString("" + this.id, (int)this.path.getCenterX() - 2, (int)this.path.getCenterY() + 2);
    }
    
    public void dibujarSelectedNodo(Graphics2D gr2d) {
        gr2d.setStroke(this.pincel);
        gr2d.setColor(Color.RED);
        gr2d.draw(this.path);
        gr2d.setColor(Color.YELLOW);
        gr2d.fillOval((int)this.path.getMinX() + 4, (int)this.path.getMinY() + 4, 28, 28);
        gr2d.setColor(Color.BLACK);
        gr2d.drawString("" + this.id, (int)this.path.getCenterX() - 2, (int)this.path.getCenterY() + 2);
    }
    
    public char getId() {
        return this.id;
    }
    
    public Point getCenter() {
        Point p = new Point();
        p.x = (int)this.path.getCenterX();
        p.y = (int)this.path.getCenterY();
        return p;
    }
    
    public int getMinX() {
        return (int)this.path.getMinX();
    }
    
    public int getMinY() {
        return (int)this.path.getMinY();
    }
    
    public int getMaxX() {
        return (int)this.path.getMaxX();
    }
    
    public int getMaxY() {
        return (int)this.path.getMaxY();
    }
    
    public void set(int x, int y) {
        this.path = new Ellipse2D.Double(x, y, 35.0D, 35.0D);
    }
    
    public boolean isSelected() {
        return this.selected;
    }
    
    public boolean contains(int xi, int yi) {
        return this.path.contains(xi, yi);
    }
    
    public void setId(char c) {
        this.id = c;
    }
}